File: strategy/python/strategies/bank_transfer_payment.py
from interfaces.payment_strategy import PaymentStrategy

class BankTransferPayment(PaymentStrategy):
    def __init__(self, account_number, bank_code):
        self.account_number = account_number
        self.bank_code = bank_code

    def pay(self, amount):
        print(f"Pagando ${amount} con Transferencia Bancaria")
        print(f"Número de cuenta: {self.account_number}")
        # Lógica de transferencia bancaria

File: strategy/php/strategies/BankTransferPayment.php
<?php

class BankTransferPayment implements PaymentStrategy {
    private $accountNumber;
    private $bankCode;

    public function __construct($accountNumber, $bankCode) {
        $this->accountNumber = $accountNumber;
        $this->bankCode = $bankCode;
    }

    public function pay($amount) {
        echo "Pagando $" . $amount . " con Transferencia Bancaria\n";
        echo "Número de cuenta: " . $this->accountNumber . "\n";
        // Lógica de transferencia bancaria
    }
}
